internal/versions/versions.go

```diff
@@ -40,15 +40,35 @@ func SupportedK8sVersions() []string {
 type ValidK8sVersion string
 
 // NewValidK8sVersion validates the given string and produces a new ValidK8sVersion object.
-func NewValidK8sVersion(k8sVersion string) (ValidK8sVersion, error) {
-	if IsSupportedK8sVersion(k8sVersion) {
-		return ValidK8sVersion(k8sVersion), nil
+// Returns an empty string if the given version is invalid.
+// strict controls whether the patch version is checked or not.
+func NewValidK8sVersion(k8sVersion string, strict bool) ValidK8sVersion {
```

Comment:
We should maybe refactor this a little in a follow up PR. Personally I'm not happy with the fact that the return type `ValidK8sVersion` can actually be an invalid (empty string) version.
A proper solution (for a separate PR) could be return an error in this function and decide how to handle that on the caller side. 
That's more of a longer term thing and shouldn't block this PR.